SwiftCSV/ParsingState.swift

```diff
@@ -20,12 +20,12 @@ struct ParsingState {
     private(set) var innerQuotes = false
 
     let delimiter: Character
-    let finishRow: () -> Void
+    let finishRow: () throws -> Void
     let appendChar: (Character) -> Void
     let finishField: () -> Void
```

Comment:
Genuine question, not a back-handed request:  What do you think about allowing all closures to throw? The public API makes sense this way. But it's a bit odd that eventually this only affects `finishRow` -- why can finishing a row fail, but not finishing a field?